app/views/results/common/_context_menu.js.erb

```diff
@@ -0,0 +1,121 @@
+<script>
+  /*
+    This file contains the Javascript code necessary for the functionality
+    of the annotation context menu. For the annotation context menu to
+    function properly, these dependencies are necessary:
+
+    Context-Menu and library related:
+      * jquery.ui-contextmenu.min.js (..\app\assets\javascripts\)
+        * Depends on jQuery and jQuery-ui
+      * context_menu.css (..\app\assets\stylesheets\)
+
+    Context-Menu button functionality:
+      * _annotations.js.erb (..\app\views\results\common\)
+
+     */
+  (function() {
+    var menu_items = {
+      new_annotation: {
+        title: 'New Annotation <kbd>Ctrl+?</kbd>',
+        cmd: 'new_annotation',
+        action: make_new_annotation,
+        disabled: true
+      },
+      pre_canned_annotations: {
+        title: 'Pre-canned <kbd>Ctrl+? <b>></b></kbd>',
+        cmd: 'precanned_annotations',
+        action: function(event, ui){ return false; },
+        children: [
+          {
+            title: "First",
+            action: function (event, ui) {
+                alert('First annot');
+            }
+          },
+          {
+            title: "Second",
+            action: "sub2",
+            tooltip: "Second"
+          }
+        ],
+        disabled: true
+      },
+      delete_annotation: {
+        title: 'Delete Annotation',
+        cmd: "delete_annotation",
+        action: function(event, ui) {
+          var clicked_element = $(ui.target);
+          var annot_id = '';
+          if (annotation_type === ANNOTATION_TYPES.CODE) {
+            $.each(clicked_element[0].attributes, function(index, attr) {
+              if (attr.nodeName.toLowerCase()
+                      .indexOf('data-annotationid') != -1) {
+
+                annot_id = attr.value;
+                return false; // breaks out of $.each iterator
+              }
+            });
+          } else {
+            annot_id = clicked_element.attr('id')
+                                      .replace('annotation_holder_', '');
+          }
+
+          if (annot_id !== '') {
+            $('.annotation_form_remove_id' + annot_id).closest("form")
```

Comment:
Eh, don't reuse the button. Refer to the route (`DELETE annotations/` with the right id) should work - you can change the dependence on `params[:submission_file_id]` in the controller method to make this work.